File: day3/FizzBuzz.cpp
#include <iostream>
#include <limits>
using namespace std;

int main() {
    int numCheck;

    while (true) {
        cout << "Please enter a number: ";

        if (cin >> numCheck) {
            for (int i = 1; i <= numCheck; i++) {
                if (i % 15 == 0) {
                    cout << "Fizzbuzz" << endl;
                } else if (i % 3 == 0) {
                    cout << "Fizz" << endl;
                } else if (i % 5 == 0) {
                    cout << "Buzz" << endl;
                } else {
                    cout << i << endl;
                }
            }
            break;
        } else {
            cout << "Please only enter a number." << endl;
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
    }

    return 0;
}

File: day3/fizz_buzz.rs
use std::io;

fn main() {
    loop {
        println!("\nPlease enter a positive whole number:");

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read input");

        match input.trim().parse::<u32>() {
            Ok(num) if num >= 1 => {
                for i in 1..=num {
                    match (i % 3, i % 5) {
                        (0, 0) => println!("FizzBuzz"),
                        (0, _) => println!("Fizz"),
                        (_, 0) => println!("Buzz"),
                        _ => println!("{}", i),
                    }
                }
                break;
            }
            _ => println!("Please only enter a positive whole number."),
        }
    }
}
